pkg/api/controller.go

```diff
@@ -2158,19 +2176,37 @@ func (c *Controller) DiffRefs(w http.ResponseWriter, r *http.Request, repository
 	if params.Type != nil && *params.Type == "two_dot" {
 		diffFunc = c.Catalog.Diff
 	}
+
+	// discern between an empty delimiter and no delimiter being passed at all
+	// by default, go-swagger will use the default value ("/") even if we pass
+	// a delimiter param that is explicitly empty. This overrides this (wrong) behavior.
```

Comment:
Well, this uncovered a weird behavior in our system. Once upon a time, there was no delimiter in the object listing API.
It worked as if there's always a delimiter that is `"/"`. When we added a configuratble one, we tried being backwards compatible. IMO, this was probably a mistake. It worked like this:

* if no delimiter is passed, assume delimiter = `/`
* if delimiter is passed but is empty (i.e. `?delimiter=`), assume no delimiter
* if delimiter is passed and is any other value, use that.

while it retained the previous behavior it's a weird choice. it means that if I add a delimiter parameter to diff endpoints I can choose between:

* doing the same thing and now assume that no delimiter actually means `/` - this would now break existing diff clients.
* make the API inconsistent by having different defaults: for objects, no delimiter means `/`, for diff it means no delimiter.
* make the API consistent by fixing object listing (and breaking it): no delimiter means no delimiter.

I went with the painful option #3.